util/auth.go

```diff
@@ -14,32 +14,40 @@
 package util
 
 import (
+	"bytes"
 	"crypto/sha1"
 	"encoding/hex"
+	"fmt"
 
 	"github.com/juju/errors"
 )
 
-// CalcPassword is the algorithm convert hashed password to auth string.
-// See https://dev.mysql.com/doc/internals/en/secure-password-authentication.html
-// SHA1( password ) XOR SHA1( "20-bytes random data from server" <concat> SHA1( SHA1( password ) ) )
-func CalcPassword(scramble, sha1pwd []byte) []byte {
-	if len(sha1pwd) == 0 {
-		return nil
-	}
-	// scrambleHash = SHA1(scramble + SHA1(sha1pwd))
-	// inner Hash
-	hash := Sha1Hash(sha1pwd)
-	// outer Hash
+// CheckScramble check scrambled password received from client.
+// The new authentication is performed in following manner:
+//   SERVER:  public_seed=create_random_string()
+//            send(public_seed)
+//   CLIENT:  recv(public_seed)
+//            hash_stage1=sha1("password")
+//            hash_stage2=sha1(hash_stage1)
+//            reply=xor(hash_stage1, sha1(public_seed,hash_stage2)
+//            // this three steps are done in scramble()
+//            send(reply)
+//   SERVER:  recv(reply)
+//            hash_stage1=xor(reply, sha1(public_seed,hash_stage2))
+//            candidate_hash2=sha1(hash_stage1)
+//            check(candidate_hash2==hash_stage2)
+//            // this three steps are done in check_scramble()
+func CheckScramble(scramble, hpwd, auth []byte) bool {
```

Comment:
Rename to `CheckScrambledPassword` is more clear.
And I think the first argument should be `salt` instead of `scramble`.